subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheMultiProjectIntegrationTest.groovy

```diff
@@ -16,8 +16,40 @@
 
 package org.gradle.configurationcache
 
+import spock.lang.Issue
+
 class ConfigurationCacheMultiProjectIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
 
+    @Issue("https://github.com/gradle/gradle/issues/18897")
+    def "picking up formerly-missing build scripts"() {
+        given:
+        settingsFile << """
+            include 'a', 'b'
+        """
+
+        def a = createDir('a') {
+            file('build.gradle') << """
+                task ok
+        """
+        }
+
+        def b = createDir('b')
+
+        and:
+        configurationCacheRun 'ok'
+
+        and:
+        b.file('build.gradle') << """
+                task ok
+        """
+
+        when:
+        configurationCacheRun 'ok'
+
+        then:
+        result.assertTasksExecuted(":a:ok", ":b:ok")
```

Comment:
🧪 I'd include an assertion on the expected cache invalidation message, `outputContains 'configuration cache cannot be reused because ...'`